### PR TITLE
propose more stimulus types

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Summary of changes

@oruebel @rly 

From discussion last week, is this the way were allow more generic types to the stimulus area of the file?

## Checklist

For all schema changes:
- [x] Add release notes for the PR to `docs/format/source/format_release_notes.rst`.

If this is the first schema change after a schema release (i.e., the version string in `core/nwb.namespace.yaml` does not
end in "-alpha"), then:
- [ ] Update the version string in `core/nwb.namespace.yaml` and `core/nwb.file.yaml` to the next major/minor/patch
  version with the suffix "-alpha". For example, if the current version is 2.4.0 and this is a minor change, then the
  new version string should be "2.5.0-alpha".
- [ ] Update the value of the `version` variable in `docs/format/source/conf.py` to the next version **without** the
  suffix "-alpha", e.g., "2.5.0".
- [ ] Update the value of the `release` variable in `docs/format/source/conf.py` to the next version **with** the suffix
  "-alpha", e.g., "2.5.0-alpha".
- [ ] Add a new section in the release notes `docs/format/source/format_release_notes.rst` for the new version
  with the date "Upcoming" in parentheses.

### CODE REVIEW

**Comment by rly** [comment]
Yes. If this is ready, I can write the corresponding pynwb PR, or you can if you want. Let me know!

**Comment by CodyCBakerPhD** [comment]
> Yes. If this is ready, I can write the corresponding pynwb PR, or you can if you want.

If you can take over this it will likely go much faster; I can certainly help write tests and give real life use cases

There's several applications we'd love to be able to utilize this in for current conversions; one for the Allen Institute, there are some visual stimuli presented described using simple `DynamicTable` types; then for holographic stimulation we may be looking into non-TimeSeries based representations more similar to LabeledEvents; I guess there's also other `ndx-events` based representations that are also considered 'stimuli' rather than just naturally observed events

File: core/nwb.file.yaml [hunk 1]
```diff
@@ -131,6 +131,12 @@ groups:
       - neurodata_type_inc: TimeSeries
         doc: TimeSeries objects containing data of presented stimuli.
         quantity: '*'
+      - neurodata_type_inc: DynamicTable
+        doc: DynamicTable objects containing data of presented stimuli.
+        quantity: '*'
+      - neurodata_type_inc: NWBDataInterface
+        doc: Generic NWB data interfaces, usually from an extension, containing data of presented stimuli.
+        quantity: '*'
```
Comment (rly) [inline core/nwb.file.yaml]:
```suggestion
        quantity: '*'
      # even though TimeSeries is a NWBDataInterface, we do not remove TimeSeries
      # in order to maintain backwards compatibility in the APIs that
      # use the neurodata_type_inc from the schema to set the variable name
```

File: core/nwb.file.yaml [hunk 2]
```diff
@@ -131,6 +131,15 @@ groups:
...
+      # even though TimeSeries is a NWBDataInterface, we do not remove TimeSeries
```
Comment (rly) [inline core/nwb.file.yaml]:
```suggestion
      # even though TimeSeries is a child type of NWBDataInterface, we do not remove TimeSeries
```

**Review by rly — APPROVED** [review]

**Pull request merged**